orttraining/orttraining/training_ops/cuda/tensor/gather_grad_impl.cu

```diff
@@ -32,10 +32,19 @@ namespace gather_grad_internal {
 // For example, given:
 //   dX_indices_sorted = [1, 1, 2, 2, 2, 3]
 //   dY_indices_sorted = [1, 4, 0, 3, 5, 2]
-// The segments will be:  '--'  '-----'  '
+// The segments will be:  '1 1'  '2 2 2'  '3'
 //
-// The segments can be processed in parallel, or further divided into partial
-// segments for increased parallelism.
+// The segments are further divided into partial segments
+// of kMaxPartialSegmentSize size for increased parallelism.
+//
+// In the above example,
+// num_segments = 3 (Number of distinct segments)
+// segment_offsets = [0, 2, 5] (Index in the sorted indices where a new segment begins)
+// per_segment_partial_segment_counts = [1, 1, 1] (Number of partial segments for each segment)
+// per_segment_partial_segment_offsets = [0, 1, 2] (Index of a partial segment where a new partial
```

Comment:
thanks for adding more comments :)
would it be "Index of a partial segment where a new segment begins"?

Reply:
You're right, I actually meant "Index of a partial segment where a new segment begins assuming the index is referring to an array comprising of elements where each element is a partial segment."
For example, if the `kMaxPartialSegmentSize` is defined as 3 and the segments are as follows:
`'1 1 1' | '2 2 2 2' | '3 3 3 3 3'`, then the partial segments array would be  `'1 1 1' | '2 2 2' | '2' | '3 3 3' | '3 3'` and the `per_segment_partial_segment_offsets` would be `[0, 1, 3]`.
I'll correct this comment.